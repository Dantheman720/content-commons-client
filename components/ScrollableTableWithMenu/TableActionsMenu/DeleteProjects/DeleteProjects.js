import React from 'react';
import PropTypes from 'prop-types';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import { Button, Modal } from 'semantic-ui-react';
import ApolloError from 'components/errors/ApolloError';
import { getCount, getPluralStringOrNot } from 'lib/utils';
import ConfirmModalContent from 'components/admin/ConfirmModalContent/ConfirmModalContent';
import DeleteProjectsList from './DeleteProjectsList/DeleteProjectsList';

const DELETE_VIDEO_PROJECTS_MUTATION = gql`
  mutation DeleteManyVideoProjects($where: VideoProjectWhereInput) {
    deleteProjects: deleteManyVideoProjects(where: $where) {
      count
    }
  }
`;

const DeleteProjects = props => {
  const {
    deleteConfirmOpen,
    handleDeleteCancel,
    handleDeleteConfirm,
    handleResetSelections,
    selections,
    showConfirmationMsg
  } = props;

  const getDrafts = () => {
    if ( selections && selections.length ) {
      return selections.filter( selection => selection.status === 'DRAFT' );
    }
    return [];
  };

  const getNonDrafts = () => {
    if ( selections && selections.length ) {
      return selections.filter( selection => selection.status !== 'DRAFT' );
    }
    return [];
  };

  return (
    <Mutation
      mutation={ DELETE_VIDEO_PROJECTS_MUTATION }
      onCompleted={ () => {
        handleResetSelections();
        handleDeleteCancel();
        showConfirmationMsg();
      } }
    >
      { ( deleteProjects, { error } ) => {
        if ( error ) return <ApolloError error={ error } />;

        const drafts = getDrafts();
        const draftsCount = getCount( drafts );

        const nonDrafts = getNonDrafts();
        const nonDraftsCount = getCount( nonDrafts );

        const hasNonDraftsOnly = draftsCount === 0 && nonDraftsCount > 0;

        return (
          <Modal className="delete" open={ deleteConfirmOpen } size="small">
            <ConfirmModalContent
              className="delete_confirm delete_confirm--video"
              headline={
                hasNonDraftsOnly
                  ? 'Only DRAFT video projects can be deleted from the dashboard.'
                  : `Are you sure you want to deleteÂ the selected DRAFT video ${getPluralStringOrNot( drafts, 'project' )}?` // eslint-disable-line
              }
            >
              { draftsCount > 0
                && (
                  <DeleteProjectsList
                    headline={ `The following DRAFT video ${getPluralStringOrNot( drafts, 'project' )} will be removed permanently from the Content Cloud:` }
                    projects={ drafts }
                    isDrafts
                  />
                ) }

              { nonDraftsCount > 0
                && (
                  <DeleteProjectsList
                    headline={ `The following non-DRAFT video ${getPluralStringOrNot( nonDrafts, 'project' )} will <strong>not</strong> be removed:` }
                    projects={ nonDrafts }
                  />
                ) }
            </ConfirmModalContent>
            <Modal.Actions>
              <Button
                content={
                  hasNonDraftsOnly
                    ? 'Take me back'
                    : 'No, take me back'
                }
                onClick={ handleDeleteCancel }
              />
              <Button
                content={
                  hasNonDraftsOnly
                    ? 'Not Applicable'
                    : 'Yes, delete forever'
                }
                disabled={ hasNonDraftsOnly }
                onClick={ () => handleDeleteConfirm( deleteProjects ) }
                outline
                primary
              />
            </Modal.Actions>
          </Modal>
        );
      } }
    </Mutation>
  );
};

DeleteProjects.propTypes = {
  deleteConfirmOpen: PropTypes.bool,
  handleDeleteCancel: PropTypes.func,
  handleDeleteConfirm: PropTypes.func,
  handleResetSelections: PropTypes.func,
  selections: PropTypes.array,
  showConfirmationMsg: PropTypes.func
};

export default DeleteProjects;

export { DELETE_VIDEO_PROJECTS_MUTATION };
