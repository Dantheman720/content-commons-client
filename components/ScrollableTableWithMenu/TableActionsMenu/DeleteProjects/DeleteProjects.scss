@import "../../../../styles/colors.scss";

.delete {
  .ui.primary.basic.button {
    color: $matisse-blue !important;
    font-weight: bold;
    box-shadow: none !important;
  
    &:hover,
    &:active,
    &:focus {
      color: #fff !important;
      box-shadow: none !important;
    }
  }
}
