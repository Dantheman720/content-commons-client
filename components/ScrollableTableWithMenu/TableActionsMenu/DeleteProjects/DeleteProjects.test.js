import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Mutation } from 'react-apollo';
import { MockedProvider } from 'react-apollo/test-utils';
import DeleteProjects, { DELETE_VIDEO_PROJECTS_MUTATION } from './DeleteProjects';

const props = {
  deleteConfirmOpen: false,
  handleDeleteCancel: jest.fn(),
  handleDeleteConfirm: jest.fn(),
  handleResetSelections: jest.fn(),
  hasSelectedAllDrafts: false,
  showConfirmationMsg: jest.fn()
};

const mocks = [
  {
    request: {
      query: DELETE_VIDEO_PROJECTS_MUTATION,
      variables: {
        where: {
          AND: [
            { id_in: ['C1', 'C2', 'C3'] },
            { status_in: ['DRAFT', 'EMBARGOED'] }
          ]
        }
      }
    },
    result: { data: { deleteProjects: 3 } }
  }
];

const Component = (
  <MockedProvider mocks={ mocks } addTypename={ false }>
    <DeleteProjects { ...props } />
  </MockedProvider>
);

describe( '<DeleteProjects />', () => {
  it( 'renders without crashing', () => {
    const wrapper = mount( Component );
    const deleteMutation = wrapper.find( 'DeleteProjects' );

    expect( deleteMutation.exists() ).toEqual( true );
    expect( toJSON( deleteMutation ) ).toMatchSnapshot();
  } );

  it( 'renders a Confirm component', () => {
    const wrapper = mount( Component );
    const deleteMutation = wrapper.find( 'DeleteProjects' );
    const confirm = deleteMutation.find( 'Confirm' );

    expect( confirm.exists() ).toEqual( true );
    expect( confirm.prop( 'className' ) ).toEqual( 'delete' );
  } );

  it( 'deleteConfirmOpen opens the Confirm component', () => {
    const newProps = { ...props, ...{ deleteConfirmOpen: true } };

    const wrapper = mount(
      <MockedProvider mocks={ mocks } addTypename={ false }>
        <DeleteProjects { ...newProps } />
      </MockedProvider>
    );

    const deleteMutation = wrapper.find( 'DeleteProjects' );
    const confirm = deleteMutation.find( 'Confirm' );

    expect( confirm.prop( 'open' ) ).toEqual( newProps.deleteConfirmOpen );
  } );

  it( 'renders the correct Confirm messages if !hasSelectedAllDrafts', () => {
    const newProps = { ...props, ...{ deleteConfirmOpen: true } };
    const wrapper = mount(
      <MockedProvider mocks={ mocks } addTypename={ false }>
        <DeleteProjects { ...newProps } />
      </MockedProvider>
    );
    const deleteMutation = wrapper.find( 'DeleteProjects' );
    const confirmModalContent = deleteMutation.find( 'ConfirmModalContent' );
    const msg1 = <p>Only selected DRAFT video project(s) will be permanently removed from the Content Cloud.</p>;
    const msg2 = <p>Selected Non-DRAFT projects will be not be removed.</p>;

    expect( confirmModalContent.contains( msg1 ) ).toEqual( true );
    expect( confirmModalContent.contains( msg2 ) ).toEqual( true );
  } );

  it( 'renders the correct Confirm message if hasSelectedAllDrafts', () => {
    const newProps = {
      ...props,
      ...{
        deleteConfirmOpen: true,
        hasSelectedAllDrafts: true
      }
    };
    const wrapper = mount(
      <MockedProvider mocks={ mocks } addTypename={ false }>
        <DeleteProjects { ...newProps } />
      </MockedProvider>
    );
    const deleteMutation = wrapper.find( 'DeleteProjects' );
    const confirmModalContent = deleteMutation.find( 'ConfirmModalContent' );
    const msg = <p>The selected DRAFT video project(s) will be removed permanently from the Content Cloud.</p>;

    expect( confirmModalContent.contains( msg ) ).toEqual( true );
  } );

  it( 'clicking the cancel button calls handleDeleteCancel', () => {
    const wrapper = mount( Component );
    const deleteMutation = wrapper.find( 'DeleteProjects' );
    const confirm = deleteMutation.find( 'Confirm' );

    confirm.prop( 'onCancel' )();
    expect( props.handleDeleteCancel ).toHaveBeenCalled();
  } );

  it( 'clicking the confirm button calls handleDeleteConfirm', () => {
    const wrapper = mount( Component );
    const deleteMutation = wrapper.find( 'DeleteProjects' );
    const confirm = deleteMutation.find( 'Confirm' );

    confirm.prop( 'onConfirm' )();
    expect( props.handleDeleteConfirm ).toHaveBeenCalled();
  } );

  it( 'mutation calls handleResetSelections, handleDeleteCancel, showConfirmationMsg on completion', () => {
    const wrapper = mount( Component );
    const deleteMutation = wrapper.find( 'DeleteProjects' );
    const mutation = deleteMutation.find( Mutation );

    mutation.prop( 'onCompleted' )();
    expect( props.handleResetSelections ).toHaveBeenCalled();
    expect( props.handleDeleteCancel ).toHaveBeenCalled();
    expect( props.showConfirmationMsg ).toHaveBeenCalled();
  } );
} );
