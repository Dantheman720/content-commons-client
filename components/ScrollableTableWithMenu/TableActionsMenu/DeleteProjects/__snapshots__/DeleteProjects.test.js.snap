// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteProjects /> renders without crashing 1`] = `
<DeleteProjects
  deleteConfirmOpen={false}
  deleteVideoProject={[Function]}
  handleActionResult={[MockFunction]}
  handleDeleteCancel={[MockFunction]}
  handleDeleteConfirm={[MockFunction]}
  handleResetSelections={[MockFunction]}
  selections={
    Array [
      Object {
        "author": Object {
          "firstName": "Edwin",
          "id": "cjrl1omfr002o075614zs8dxz",
          "lastName": "Mah",
        },
        "categories": Array [
          Object {
            "id": "cjubblzwq15r10756m11zntim",
            "translations": Array [
              Object {
                "id": "cjubbleh10v2607568m3kisep",
                "language": Object {
                  "id": "cjsq439dz005607560gwe7k3m",
                  "locale": "en-us",
                },
                "name": "environment",
              },
              Object {
                "id": "cjubblehi0v2e0756ase6olan",
                "language": Object {
                  "id": "cjsuxnpv700dl0756wg4adr6k",
                  "locale": "es-es",
                },
                "name": "medioambiente",
              },
              Object {
                "id": "cjubblehw0v2m075606oy0x39",
                "language": Object {
                  "id": "cjsq4565v005c0756f0lqbfe4",
                  "locale": "fr-fr",
                },
                "name": "Environnement",
              },
              Object {
                "id": "cjubbleic0v2u075618dkevjf",
                "language": Object {
                  "id": "cjt5qc65o04020756oius55mh",
                  "locale": "pt-br",
                },
                "name": "meio ambiente",
              },
              Object {
                "id": "cjubbleix0v320756q1xjh9pj",
                "language": Object {
                  "id": "cjsq47fum005i0756qlhc0wta",
                  "locale": "ar",
                },
                "name": "البيئة",
              },
              Object {
                "id": "cjubblejc0v3a0756phz2der5",
                "language": Object {
                  "id": "cjubbldhj0uqh0756pco0zk2r",
                  "locale": "fa-ir",
                },
                "name": "محیط زیست",
              },
              Object {
                "id": "cjubblejq0v3i0756lfud6sqg",
                "language": Object {
                  "id": "cjsq4967o005o0756gvpl1f5k",
                  "locale": "zh-cn",
                },
                "name": "环境",
              },
              Object {
                "id": "cjubblek60v3q07566bqtlpif",
                "language": Object {
                  "id": "cjt06xh0e02q70756dnuo3g8e",
                  "locale": "ru-ru",
                },
                "name": "Окружающая среда",
              },
              Object {
                "id": "cjubblekm0v3y0756o0gk6nou",
                "language": Object {
                  "id": "cjt5qviod040v0756m5yhxs05",
                  "locale": "id-id",
                },
                "name": "lingkungan hidup",
              },
              Object {
                "id": "cjubblel30v460756humkwrmu",
                "language": Object {
                  "id": "cjt5qnpem040e07567msmql7j",
                  "locale": "ur",
                },
                "name": "ماحول",
              },
            ],
          },
        ],
        "createdAt": "2019-06-17T15:23:46.184Z",
        "id": "cjx0j3ts9066y0708k1hg2g1b",
        "projectTitle": "Philippines Village",
        "status": "DRAFT",
        "team": Object {
          "id": "cjrkzhvku000f0756l44blw33",
          "name": "IIP Video Production",
          "organization": "Department of State",
        },
        "thumbnails": Array [
          Object {
            "alt": null,
            "id": "cjx0j3xqz067z0708frd7sd9g",
            "url": "2019/06/cjx0j3ts9066y0708k1hg2g1b/coverr-philippines-village-views-1559889358075.jpg",
          },
        ],
        "updatedAt": "2019-06-17T15:23:46.184Z",
        "visibility": "PUBLIC",
      },
      Object {
        "author": Object {
          "firstName": "Edwin",
          "id": "cjrl1omfr002o075614zs8dxz",
          "lastName": "Mah",
        },
        "categories": Array [
          Object {
            "id": "cjubblzw915qe07560zkz9xas",
            "translations": Array [
              Object {
                "id": "cjubble530uxq0756q5bphm9c",
                "language": Object {
                  "id": "cjsq439dz005607560gwe7k3m",
                  "locale": "en-us",
                },
                "name": "economic opportunity",
              },
              Object {
                "id": "cjubble5n0uxy0756q8qnn3cq",
                "language": Object {
                  "id": "cjsuxnpv700dl0756wg4adr6k",
                  "locale": "es-es",
                },
                "name": "Oportunidades económicas",
              },
              Object {
                "id": "cjubble640uy607568seiatuw",
                "language": Object {
                  "id": "cjsq4565v005c0756f0lqbfe4",
                  "locale": "fr-fr",
                },
                "name": "Opportunités économiques",
              },
              Object {
                "id": "cjubble6p0uye0756h6aw02vm",
                "language": Object {
                  "id": "cjt5qc65o04020756oius55mh",
                  "locale": "pt-br",
                },
                "name": "Oportunidade econômica",
              },
              Object {
                "id": "cjubble7c0uym0756vp6lcupx",
                "language": Object {
                  "id": "cjsq47fum005i0756qlhc0wta",
                  "locale": "ar",
                },
                "name": "فرص اقتصادية",
              },
              Object {
                "id": "cjubble800uyu07563abeywer",
                "language": Object {
                  "id": "cjubbldhj0uqh0756pco0zk2r",
                  "locale": "fa-ir",
                },
                "name": "فرصت های اقتصادی",
              },
              Object {
                "id": "cjubble8k0uz20756wgu91l9d",
                "language": Object {
                  "id": "cjsq4967o005o0756gvpl1f5k",
                  "locale": "zh-cn",
                },
                "name": "经济机会",
              },
              Object {
                "id": "cjubble930uza0756rmnl6oij",
                "language": Object {
                  "id": "cjt06xh0e02q70756dnuo3g8e",
                  "locale": "ru-ru",
                },
                "name": "Экономические возможности",
              },
              Object {
                "id": "cjubblea00uzi0756ciw0vz1j",
                "language": Object {
                  "id": "cjt5qviod040v0756m5yhxs05",
                  "locale": "id-id",
                },
                "name": "Peluang  Ekonomi",
              },
              Object {
                "id": "cjubbleak0uzq0756l1x17v3p",
                "language": Object {
                  "id": "cjt5qnpem040e07567msmql7j",
                  "locale": "ur",
                },
                "name": "اقتصادی موقع",
              },
            ],
          },
          Object {
            "id": "cjubblzwq15r10756m11zntim",
            "translations": Array [
              Object {
                "id": "cjubbleh10v2607568m3kisep",
                "language": Object {
                  "id": "cjsq439dz005607560gwe7k3m",
                  "locale": "en-us",
                },
                "name": "environment",
              },
              Object {
                "id": "cjubblehi0v2e0756ase6olan",
                "language": Object {
                  "id": "cjsuxnpv700dl0756wg4adr6k",
                  "locale": "es-es",
                },
                "name": "medioambiente",
              },
              Object {
                "id": "cjubblehw0v2m075606oy0x39",
                "language": Object {
                  "id": "cjsq4565v005c0756f0lqbfe4",
                  "locale": "fr-fr",
                },
                "name": "Environnement",
              },
              Object {
                "id": "cjubbleic0v2u075618dkevjf",
                "language": Object {
                  "id": "cjt5qc65o04020756oius55mh",
                  "locale": "pt-br",
                },
                "name": "meio ambiente",
              },
              Object {
                "id": "cjubbleix0v320756q1xjh9pj",
                "language": Object {
                  "id": "cjsq47fum005i0756qlhc0wta",
                  "locale": "ar",
                },
                "name": "البيئة",
              },
              Object {
                "id": "cjubblejc0v3a0756phz2der5",
                "language": Object {
                  "id": "cjubbldhj0uqh0756pco0zk2r",
                  "locale": "fa-ir",
                },
                "name": "محیط زیست",
              },
              Object {
                "id": "cjubblejq0v3i0756lfud6sqg",
                "language": Object {
                  "id": "cjsq4967o005o0756gvpl1f5k",
                  "locale": "zh-cn",
                },
                "name": "环境",
              },
              Object {
                "id": "cjubblek60v3q07566bqtlpif",
                "language": Object {
                  "id": "cjt06xh0e02q70756dnuo3g8e",
                  "locale": "ru-ru",
                },
                "name": "Окружающая среда",
              },
              Object {
                "id": "cjubblekm0v3y0756o0gk6nou",
                "language": Object {
                  "id": "cjt5qviod040v0756m5yhxs05",
                  "locale": "id-id",
                },
                "name": "lingkungan hidup",
              },
              Object {
                "id": "cjubblel30v460756humkwrmu",
                "language": Object {
                  "id": "cjt5qnpem040e07567msmql7j",
                  "locale": "ur",
                },
                "name": "ماحول",
              },
            ],
          },
        ],
        "createdAt": "2019-06-17T19:42:42.188Z",
        "id": "cjx0sctfx06ry0708bmtuf37f",
        "projectTitle": "Philippines Village 2",
        "status": "DRAFT",
        "team": Object {
          "id": "cjrkzhvku000f0756l44blw33",
          "name": "IIP Video Production",
          "organization": "Department of State",
        },
        "thumbnails": Array [
          Object {
            "alt": null,
            "id": "cjx0sd0yy06t30708by0b4suo",
            "url": "2019/06/cjx0sctfx06ry0708bmtuf37f/coverr-philippines-village-views-1559889358075.jpg",
          },
        ],
        "updatedAt": "2019-06-17T19:42:42.188Z",
        "visibility": "PUBLIC",
      },
      Object {
        "author": Object {
          "firstName": "Jane",
          "id": "cjuybpucc1tuk0756bpi84enp",
          "lastName": "Doe",
        },
        "categories": Array [],
        "createdAt": "2019-06-17T15:35:31.065Z",
        "id": "cjx0jixoa06c60708z2i7t2t3",
        "projectTitle": "Made in America 14",
        "status": "PUBLISHED",
        "team": Object {
          "id": "cjrkzhvku000f0756l44blw33",
          "name": "IIP Video Production",
          "organization": "Department of State",
        },
        "thumbnails": Array [],
        "updatedAt": "2019-06-17T15:35:31.065Z",
        "visibility": "PUBLIC",
      },
    ]
  }
  showConfirmationMsg={[MockFunction]}
>
  <Modal
    centered={true}
    className="delete"
    closeOnDimmerClick={true}
    closeOnDocumentClick={false}
    dimmer={true}
    eventPool="Modal"
    open={false}
    size="small"
  >
    <Portal
      closeOnDocumentClick={false}
      closeOnEscape={true}
      eventPool="Modal"
      mountNode={<body />}
      onClose={[Function]}
      onMount={[Function]}
      onOpen={[Function]}
      onUnmount={[Function]}
      open={false}
      openOnTriggerClick={true}
    />
  </Modal>
</DeleteProjects>
`;
